runner/ollamarunner/runner.go

```
@@ -591,18 +593,24 @@ func (s *Server) completion(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var grammar *sample.Grammar
+	var err error
+	if req.Grammar != "" {
+		grammar, err = sample.NewGrammar(s.vocab, req.Grammar)
+		if err != nil {
+			slog.Error("failed to load vocab for sampling", "error", err)
```

This should probably return an http error for now